guiEditor/src/components/propertyTab/propertyTabComponent.tsx

```diff
@@ -126,15 +127,18 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
     }
 
     save() {
+        this.props.globalState.guiTexture.removeControl(this.props.globalState.workbench.artBoardBackground);
         try {
             const json = JSON.stringify(this.props.globalState.guiTexture.serializeContent());
             StringTools.DownloadAsFile(this.props.globalState.hostDocument, json, "guiTexture.json");
         } catch (error) {
             alert("Unable to save your GUI");
         }
+        this.props.globalState.guiTexture.getChildren()[0].children.unshift(this.props.globalState.workbench.artBoardBackground);
     }
 
     saveToSnippetServer() {
+        this.props.globalState.guiTexture.removeControl(this.props.globalState.workbench.artBoardBackground);

```

Comment:
Same here and 189 ?

Reply:
Also this maybe could be factorize with a function taking a callback in parameter as you use the same pattern twice ?

Reply:
Not sure what you mean by this. Could you explain?

Reply:
```
function foo(doSmthgCallback: () => void) {
  this.props.globalState.guiTexture.removeControl(this.props.globalState.workbench.artBoardBackground);
  doSmthgCallback();
  adt.getChildren()[0].children.unshift(this.props.globalState.workbench.artBoardBackground);
}
```

smthg in this spirit ???